### PR TITLE
Issue #11865: Jakarta'ize appSecurityClient-1.0 feature and add FATs. Update …

### LINKED ISSUE #13131: JEE9: Check if the BeanManager is correctly set into the ServletContext
In AbstractInitialListenerRegistration, line 92, we set the BeanManager into the ServletContext...

                //Put bean manager on the servlet context
                isc.setAttribute("javax.enterprise.inject.spi.BeanManager", beanManager);

I wonder if that key is still correct. Should we remove that hard-coded string in favour of `BeanManager.class.getName()`

### PR DESCRIPTION
**Pull request by jvanhill**
…Jakarta security FATs to run EE9. Resolved issue #13131.

fixes #13131

### CODE REVIEW

**Comment by jvanhill** [comment]
#libby

**Comment by jvanhill** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurityClient.feature
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurityClient.feature:10]: does `com.ibm.ws.cdi.client` bundle need to be transformed?

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurity.feature
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurity.feature:8]: does `com.ibm.ws.cdi.security` need to be transformed?

File: dev/com.ibm.ws.security.client_fat/.classpath
Comment (tevans78) [inline dev/com.ibm.ws.security.client_fat/.classpath:1]: New projects should be named `io.openliberty...`. In this case `io.openliberty.security.internal.client_fat` or maybe `io.openliberty.security.client.internal_fat`?

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurityClient.feature
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurityClient.feature:10]: No. It does not depend on any Java EE classes.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurity.feature
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurity.feature:8]: No. It does not depend on any Java EE classes.

File: dev/com.ibm.ws.security.client_fat/.classpath
Comment (jvanhill) [reply dev/com.ibm.ws.security.client_fat/.classpath:1]: Technically, this is a project that I moved from WS-CD-Open. Additionally, I name the fats to match the project they are testing. In this case, the project being tested is `com.ibm.ws.security.client`.

**Comment by skasund** [comment]
L2 message review not required

File: dev/com.ibm.websphere.appserver.features/visibility/public/appSecurityClient-2.0/resources/l10n/io.openliberty.appSecurityClient-2.0.properties
Comment (dmuelle) [inline dev/com.ibm.websphere.appserver.features/visibility/public/appSecurityClient-2.0/resources/l10n/io.openliberty.appSecurityClient-2.0.properties:18]: needs a serial comma

This feature enables application security, CSIv2, and SSL on the client.

**Comment by jvanhill** [comment]
It appears with the renaming of the `com.ibm.websphre.appserver.javax.servlet-3.0/3.1/4.0` and `io.openliberty.jakarta.servlet-5.0` features to `io.openliberty.servlet.api-3.0/3.1/4.0/5.0` issue #13242 that we can avoid having to create a new appSecurityClient-2.0 feature as we can use auto features to provision the differing bundles.

I am spinning a build at this time to verify that I haven't broken any other FATs, but will need to wait until #13242 has been merged to remove the appSecurityClient-2.0 feature from this PR.

#build

**Comment by jvanhill** [comment]
#libby

File: dev/com.ibm.websphere.appserver.features/visibility/public/appSecurityClient-2.0/resources/l10n/io.openliberty.appSecurityClient-2.0.properties
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/public/appSecurityClient-2.0/resources/l10n/io.openliberty.appSecurityClient-2.0.properties]: Updated the text.

**Review by dmuelle — APPROVED** [review]
LGTM

**Comment by jvanhill** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurityClient.feature
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurityClient.feature:1]: The javaee version of this autofeature had 1.0 after appSecurityClient.  Did you remove it on purpose?

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurity.feature
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurity.feature:1]: The javaee version of this autofeature had 1.0 after appSecurity. Did you remove it on purpose?  Should it take on 4.0?

File: dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.csiv2-2.0.feature
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.csiv2-2.0.feature:1]: I don't see any reason that this auto feature cannot be changed to io.openliberty.  It is not singleton feature.  It is funny that is is named csiv2 in reality since the auto feature doesn't actually rely on a csiv2 feature being enabled, but rather enterpriseBeansRemote and appSecurity

File: dev/com.ibm.ws.cdi.web/src/com/ibm/ws/cdi/web/impl/AbstractInitialListenerRegistration.java
Comment (jhanders34) [inline dev/com.ibm.ws.cdi.web/src/com/ibm/ws/cdi/web/impl/AbstractInitialListenerRegistration.java:92]: getName() is a much simpler method to call and there is no reason to require getCanonicalName

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureDefinitionUtils.java
Comment (jhanders34) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureDefinitionUtils.java:89]: There is no appSecurityClient-2.0 feature

File: dev/com.ibm.ws.security.client_fat/build.gradle
Comment (jhanders34) [inline dev/com.ibm.ws.security.client_fat/build.gradle:31]: Shouldn't some of these applications be built automatically with the tooling without having to write them all out in the gradle file?

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImpl.java
Comment (jhanders34) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImpl.java:489]: The OR of the conditional will never be triggered, so is there a point in having it?

File: dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd
Comment (jhanders34) [inline dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd:1]: Will this be used on the commercial liberty repo still?  If so you will probably want to set the publish attribute so a stub project is created over there pointing to it on the open liberty side.

File: dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd
Comment (jhanders34) [inline dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd:36]: Why using a binary here instead of depending on a maven / dhe repository location?

File: dev/com.ibm.ws.security.client_fat/build.gradle
Comment (jhanders34) [inline dev/com.ibm.ws.security.client_fat/build.gradle:259]: Not sure why these binaries are included in the application or put into github.  The APIs are provided by the liberty runtime.   I suggest removing them from github and not be included in the ear file if possible.

**Review by jhanders34 — CHANGES_REQUESTED** [review]

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurityClient.feature
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurityClient.feature:1]: Yes. I didn't want to tie it to any version of appSecurityClient, just that app security is enabled.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurity.feature
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-appSecurity.feature:1]: Yes. I didn't want to tie it to any version of appSecurity, just that app security is enabled.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.csiv2-2.0.feature
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.csiv2-2.0.feature:1]: It could be renamed. I only kept it named similar to the original b/c I couldn't rename that one and I wanted them to be paired. If you want me to rename with io.openliberty i can. I believe the CSIV2 is probably indicative of the functionality that is being enabled since remote EJBs required CSIv2.

File: dev/com.ibm.ws.cdi.web/src/com/ibm/ws/cdi/web/impl/AbstractInitialListenerRegistration.java
Comment (jvanhill) [reply dev/com.ibm.ws.cdi.web/src/com/ibm/ws/cdi/web/impl/AbstractInitialListenerRegistration.java:92]: Changed.

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureDefinitionUtils.java
Comment (jvanhill) [reply dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureDefinitionUtils.java:89]: Changed.

I had originally created one before the private servlet feature rename. Had missed this change from prior to merging in those changes.

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImpl.java
Comment (jvanhill) [reply dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImpl.java:489]: Good catch. I can remove it.

File: dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd
Comment (jvanhill) [reply dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd:1]: Do I need to set this?

`publish.wlp.jar.disabled: false`

File: dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd
Comment (jvanhill) [reply dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd:36]: They made some changes to this JAR, I am not entirely certain what those changes were. I pulled in the one they were using over in WS.

File: dev/com.ibm.ws.security.client_fat/build.gradle
Comment (jvanhill) [reply dev/com.ibm.ws.security.client_fat/build.gradle:259]: When reviewing with Ryan, I realized these were here. Going to remove them.

File: dev/com.ibm.ws.security.client_fat/build.gradle
Comment (jvanhill) [reply dev/com.ibm.ws.security.client_fat/build.gradle:31]: Is there an easy way to do this? If so, I am more than happy to make the changes.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.csiv2-2.0.feature
Comment (jhanders34) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.csiv2-2.0.feature:1]: In general the direction is for new features to be io.openliberty if possible.  We are not trying to do renames of existing features.

File: dev/com.ibm.ws.security.client_fat/build.gradle
Comment (jhanders34) [reply dev/com.ibm.ws.security.client_fat/build.gradle:31]: I thought you wanted to do it proactively during build and bloat the autoFVT.zip file instead of during execution of the tests.  I don't know if you can do that without what you are doing.  @aguibert would probably know more.

File: dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd
Comment (jhanders34) [reply dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd:1]: yes if it is needed both other fats outside of open liberty

File: dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd
Comment (jhanders34) [reply dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd:36]: You can still publish a version of it into the dhe / artifactory repo.

File: dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd
Comment (jvanhill) [reply dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd:1]: Updated the bnd.bnd file.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.csiv2-2.0.feature
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.csiv2-2.0.feature:1]: Renamed it `io.openliberty.enterpriseBeansRemote-appSecurity-1.0`

**Review by ryanesch — APPROVED** [review]

File: dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd
Comment (jvanhill) [reply dev/com.ibm.ws.security_fat.common.tooling/bnd.bnd]: I removed this project since parts of it had been created already.

File: dev/com.ibm.ws.security.client_fat/build.gradle
Comment (jvanhill) [reply dev/com.ibm.ws.security.client_fat/build.gradle:30]: I left them in the build.gradle, as we do with all our other projects.

**Comment by jvanhill** [comment]
#build
#libby

**Comment by jvanhill** [comment]
#build
#libby

**Comment by jvanhill** [comment]
#build

File: dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java
Comment (jhanders34) [inline dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java:63]: I think you want this to be mail not jakartaMail

File: dev/com.ibm.ws.security.client_fat/fat/src/com/ibm/ws/security/client/fat/JavaColonInjectionsTest.java
Comment (jhanders34) [inline dev/com.ibm.ws.security.client_fat/fat/src/com/ibm/ws/security/client/fat/JavaColonInjectionsTest.java:38]: This comment should say mail-2.0.  It might be available now.

**Review by jhanders34 — CHANGES_REQUESTED** [review]
I have put in a few comments.  If you only make changes related to this I should be able to approve rather quickly.  Since there is a conflict with the stream right now I would suggest doing a rebase as well and resolve the conflict before doing another PB.

File: dev/com.ibm.ws.security.client_fat/fat/src/com/ibm/ws/security/client/fat/JavaColonInjectionsTest.java
Comment (jhanders34) [reply dev/com.ibm.ws.security.client_fat/fat/src/com/ibm/ws/security/client/fat/JavaColonInjectionsTest.java:38]: I just looked and it is edition = beta so it would appears available.

File: dev/cnf/oss_dependencies.maven
Comment (jvanhill) [reply dev/cnf/oss_dependencies.maven:473]: Removed.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.enterpriseBeansRemote-appSecurity-1.0.feature
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.enterpriseBeansRemote-appSecurity-1.0.feature:2]: I originally added -1.0 to indicate the first version of the feature, not the version of any feature that provisions this feature. I will remove the -1.0 from the end of the feature / symbolic name.

File: dev/fattest.simplicity/src/componenttest/rules/repeater/JakartaEE9Action.java
Comment (jvanhill) [reply dev/fattest.simplicity/src/componenttest/rules/repeater/JakartaEE9Action.java:77]: Made the change.

File: dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java
Comment (jvanhill) [reply dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java:63]: Made the change.

File: dev/com.ibm.ws.security.client_fat/fat/src/com/ibm/ws/security/client/fat/JavaColonInjectionsTest.java
Comment (jvanhill) [reply dev/com.ibm.ws.security.client_fat/fat/src/com/ibm/ws/security/client/fat/JavaColonInjectionsTest.java]: Removed the skip for repeat as mail-2.0 is available. I also had to update `wlp-jakartaee-transform/rules/jakarta-renames.properties` (rename javax.mail to jakarta.mail) and `com.ibm.ws.security.client_fat/bnd.bnd`  (add EE9 tested features) for this change.

**Comment by jvanhill** [comment]
#libby

**Review by jhanders34 — APPROVED** [review]

File: dev/wlp-jakartaee-transform/rules/jakarta-xml-dd.properties
Comment (jhanders34) [inline dev/wlp-jakartaee-transform/rules/jakarta-xml-dd.properties:46]: This particular one was moved earlier in the file to line up with other ones.  You may want to remove the duplicate.

Comment (jvanhill) [reply dev/wlp-jakartaee-transform/rules/jakarta-xml-dd.properties]: Moved this up.

**Comment by jvanhill** [comment]
#libby
#build

**Comment by jvanhill** [comment]
There was a single JCA failure, but it was a preexisting failure that has a build break defect open for it. There were some UI failures as well that were unrelated.

**Pull request merged**
